From b95a7a245cf0bbb9a832874bc43e5e935aa771c5 Mon Sep 17 00:00:00 2001
From: Ramax Lo <ramax.lo@mediatek.com>
Date: Thu, 20 Feb 2025 10:41:51 +0800
Subject: [PATCH] GENIO: drivers,lib: Strip path components

The host tools pnmtologo, conmakehash and build_OID_registry output
absolute paths in the generated files, which causes warnings of the
package QA process of Yocto. This commit modifies the source to only
output basenames of paths.

Signed-off-by: Ramax Lo <ramax.lo@mediatek.com>
---
 drivers/tty/vt/conmakehash.c   | 6 +++++-
 drivers/video/logo/pnmtologo.c | 7 ++++++-
 lib/build_OID_registry         | 3 ++-
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/drivers/tty/vt/conmakehash.c b/drivers/tty/vt/conmakehash.c
index cddd789fe46e..89474be2a769 100644
--- a/drivers/tty/vt/conmakehash.c
+++ b/drivers/tty/vt/conmakehash.c
@@ -16,6 +16,7 @@
 #include <sysexits.h>
 #include <string.h>
 #include <ctype.h>
+#include <libgen.h>
 
 #define MAX_FONTLEN 256
 
@@ -82,6 +83,7 @@ int main(int argc, char *argv[])
   int i, nuni, nent;
   int fp0, fp1, un0, un1;
   char *p, *p1;
+  char tmpbuf[65536];
 
   if ( argc < 2 || argc > 5 )
     usage(argv[0]);
@@ -242,6 +244,8 @@ int main(int argc, char *argv[])
   for ( i = 0 ; i < fontlen ; i++ )
     nuni += unicount[i];
 
+  strncpy(tmpbuf, argv[1], sizeof(tmpbuf) - 1);
+  tmpbuf[sizeof(tmpbuf) - 1] = '\0';
   printf("\
 /*\n\
  * Do not edit this file; it was automatically generated by\n\
@@ -253,7 +257,7 @@ int main(int argc, char *argv[])
 #include <linux/types.h>\n\
 \n\
 u8 dfont_unicount[%d] = \n\
-{\n\t", argv[1], fontlen);
+{\n\t", basename(tmpbuf), fontlen);
 
   for ( i = 0 ; i < fontlen ; i++ )
     {
diff --git a/drivers/video/logo/pnmtologo.c b/drivers/video/logo/pnmtologo.c
index ada5ef6e51b7..9b57ec210ab1 100644
--- a/drivers/video/logo/pnmtologo.c
+++ b/drivers/video/logo/pnmtologo.c
@@ -13,6 +13,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <libgen.h>
 
 
 static const char *programname;
@@ -223,6 +224,8 @@ static inline int is_equal(struct color c1, struct color c2)
 
 static void write_header(void)
 {
+	char tmp[65536];
+
 	/* open logo file */
 	if (outputname) {
 		out = fopen(outputname, "w");
@@ -232,10 +235,12 @@ static void write_header(void)
 		out = stdout;
 	}
 
+	strncpy(tmp, filename, sizeof(tmp) - 1);
+	tmp[sizeof(tmp) - 1] = '\0';
 	fputs("/*\n", out);
 	fputs(" *  DO NOT EDIT THIS FILE!\n", out);
 	fputs(" *\n", out);
-	fprintf(out, " *  It was automatically generated from %s\n", filename);
+	fprintf(out, " *  It was automatically generated from %s\n", basename(tmp));
 	fputs(" *\n", out);
 	fprintf(out, " *  Linux logo %s\n", logoname);
 	fputs(" */\n\n", out);
diff --git a/lib/build_OID_registry b/lib/build_OID_registry
index d7fc32ea8ac2..25345ca9ba06 100755
--- a/lib/build_OID_registry
+++ b/lib/build_OID_registry
@@ -8,6 +8,7 @@
 #
 
 use strict;
+use File::Basename;
 
 my @names = ();
 my @oids = ();
@@ -35,7 +36,7 @@ close IN_FILE || die;
 #
 open C_FILE, ">$ARGV[1]" or die;
 print C_FILE "/*\n";
-print C_FILE " * Automatically generated by ", $0, ".  Do not edit\n";
+print C_FILE " * Automatically generated by ", basename($0), ".  Do not edit\n";
 print C_FILE " */\n";
 
 #
-- 
2.45.2

